pulpcore/cli/rpm/comps.py

```diff
@@ -0,0 +1,56 @@
+import gettext
+from typing import IO, Optional
+
+import click
+
+from pulpcore.cli.common.context import (
+    PluginRequirement,
+    PulpContext,
+    pass_entity_context,
+    pass_pulp_context,
+)
+from pulpcore.cli.rpm.context import PulpRpmCompsXmlContext, PulpRpmRepositoryContext
+
+_ = gettext.gettext
+
+
+@click.group()
+@click.option(
+    "-t",
+    "--type",
+    "repo_type",
+    type=click.Choice(["rpm"], case_sensitive=False),
+    default="rpm",
+)
+@pass_pulp_context
+@click.pass_context
+def comps(ctx: click.Context, pulp_ctx: PulpContext, repo_type: str) -> None:
+    if repo_type == "rpm":
+        pulp_ctx.needs_plugin(PluginRequirement("rpm", min="3.17.0dev"))
+        ctx.obj = PulpRpmCompsXmlContext(pulp_ctx)
+    else:
+        raise NotImplementedError()
+
+
+@comps.command()
+@click.option("--file", type=click.File("rb"), required=True)
+@click.option("--repository", type=str)
```

Comment:
You should turn this into a `resource_option`.

Reply:
:+1: